src/hotspot/share/gc/shenandoah/shenandoahMark.inline.hpp

```diff
@@ -365,6 +366,9 @@ inline void ShenandoahMark::mark_ref(ShenandoahObjToScanQueue* q,
     marked = mark_context->mark_strong(obj, /* was_upgraded = */ skip_live);
   }
   if (marked) {
+    if (ObjectCountClosure::should_send_event<EventObjectCountAfterGC>()) {
```

Comment:
Record the performance impact of having this check with the event disabled to measure the impact of the if test.

If it turns out to be substantial, consider templatizing the `mark_ref` method. (While bearing in mind Knuth's admonition that "premature optimization is the root of all evil" -- so definitely first measure before you change! And when you measure, please record in your log-book/work-book/journal, so you can use it in your report/presentation.)